### PR TITLE
refactor(handlers,types): separate GUILD_CREATE fields from Guild object

### PR DESCRIPTION
**Pull request by lts20050703**
Closes: https://github.com/discordeno/discordeno/issues/2213
Upstream: https://github.com/discord/discord-api-docs/commit/78a152e

types: Add DiscordGuildCreate interface
transformers: Create GuildCreateToggles and transformGuildCreate transformer
handlers: handleGuildCreate use the transformGuildCreate transformer

### CODE REVIEW

File: transformers/guildCreate.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import { Bot } from "../bot.ts";
+import { Collection } from "../util/collection.ts";
+import { DiscordGuildCreate } from "../types/discord.ts";
+import { Optionalize } from "../types/shared.ts";
+import { GuildCreateToggles } from "./toggles/guild.ts";
+
+export function transformGuildCreate(bot: Bot, payload: { guild: DiscordGuildCreate } & { shardId: number }) {
+  const guildId = bot.transformers.snowflake(payload.guild.id);
+
+  const guild = {
```
Comment (itohatweb) [inline transformers/guildCreate.ts:11]: Please do `...transformGuild(payload)` and then just add the other fields

File: transformers/toggles/guild.ts [hunk 2]
```diff
@@ -94,6 +92,15 @@ export class GuildToggles extends ToggleBitfieldBigint {
...
   }
+}
+
+export class GuildCreateToggles extends GuildToggles {
+  constructor(guild: DiscordGuildCreate) {
+    super(guild);
+
+    if (guild.large) this.add(GuildToggle.large);
+    if (guild.unavailable) this.add(GuildToggle.unavailable);
+  }
```
Comment (itohatweb) [inline transformers/toggles/guild.ts:103]: Move to separate file

**Review by itohatweb — CHANGES_REQUESTED** [review]
Please revert the unrelated format changes

**Review by itohatweb — APPROVED** [review]

**Comment by Skillz4Killz** [comment]
Not a fan of this PR. I don't want to go this route where we end up having like 20 different transformers for a channel because each has a few different props on it.

**Comment by itohatweb** [comment]
We won't do that for channels, since TypeScript does not provide anything what we could use and still maintain a good UX.
But it is different in this case. Why not say through types that eg. `channels` never exists when you use the `getGuild` helper.

**Comment by Skillz4Killz** [comment]
That can be done with types without having to create multiple transformers for each type.

```ts
function getGuild() {
  return {} as SpecialGuildInterface;
}
```

**Pull request closed without merging**